Wikipedia/Code/NSLocale+WMFExtras.swift

```diff
@@ -120,21 +120,25 @@ extension NSLocale {
             return Locale.autoupdatingCurrent
         }
         
-        let languageInfo = MWLanguageInfo(forCode: language)
-        let code = languageInfo.code
+        // Previous code took the wikipediaLanguage value and called a method
+        // that would return the same language code that was passed in
+        // unless 'simple' or 'test' was passed in instead of a language code
+        // in which case "en" would be returned. This asserion double checks that
+        // that old substituion is no longer being relied upon.
+        assert(language != "simple" && language != "test")
```

Comment:
I see this assertion getting hit a couple of times when I follow these steps:

1. Add "Simple English" wiki to languages via app settings.
2. Assertion failure is hit whenever Explore feed tries to load. Although if you don't reach this you probably will reach it when digging into the article's edit history on an article in Simple.
3. Add "Test" wiki to languages via app settings.
4. Search for any article in Test. Go to article. Go down to footer and tap "View edit history".

Reply:
I did not know there was a Simple English version of Wikipedia.

I've updated and removed the assert and added a new conditional that uses the "en" locale if the incoming language is "simple" or "test".

This way, once the locales are cached once, we don't need to keep checking if the language string is "simple" or "test". (Which we were doing before these changes.)

The other use cases in where this transformation happened before the refactor were both checking if the incoming language string was in a set of RTL language codes. This will work with 'simple' and 'test' as-is without any special transformation to 'en'.

I've also added a comment describing why the special casing is necessary.